packages/java-parser/src/productions/types-values-and-variables.js

```diff
@@ -136,12 +136,15 @@ function defineRules($, t) {
     });
     $.CONSUME(t.LSquare);
     $.CONSUME(t.RSquare);
-    $.MANY2(() => {
-      $.MANY3(() => {
-        $.SUBRULE2($.annotation);
-      });
-      $.CONSUME2(t.LSquare);
-      $.CONSUME2(t.RSquare);
+    $.MANY2({
+      GATE: () => $.BACKTRACK_LOOKAHEAD($.isDims),
```

Comment:
this GATE seems too restrictive, what if it starts with annotation?

Reply:
I don't really understand: if it start with annotations, the Gate will be OK, as the `isDims` rule start with MANY(annotation). Or am I missing something? The name isDims is maybe a bit misleading

Reply:
I don't really understand: if it start with annotations, the Gate will be OK, as the `isDims` rule start with MANY(annotation). Or am I missing something? The name isDims is maybe a bit misleading

Reply:
I missed that isDims also checks annotations